components/NewTripForm/index.js

```diff
@@ -1,12 +1,49 @@
 import styled from "styled-components";
 import { trips } from "../../lib/data";
 import { useRouter } from "next/router";
+import { useState } from "react";
 
 export default function NewTripForm() {
-  // default value till flexible trip duration will be implemented
-  const tripDurationInDays = 3;
   const router = useRouter();
 
+  const [startDate, setStartDate] = useState("");
+  const [endDate, setEndDate] = useState("");
+  const [tripDurationInDays, setTripDurationInDays] = useState(0);
+
+  // handler for the onChange attribute of "start-date" input field
+  function handleStartDateChange(event) {
+    const startDateValue = event.target.value;
+    setStartDate(startDateValue);
+    calculateTripDuration(startDateValue, endDate); //the second argument "endDate" is the useState value
+  }
+  // handler for the onChange attribute of "end-date" input field
+  function handleEndDateChange(event) {
+    const endDateValue = event.target.value;
+    setEndDate(endDateValue);
+    calculateTripDuration(startDate, endDateValue); // the first argument "startDate" is the useState value
+  }
+
+  // calculate trip duration in days based on the input values of the form fields "start-date" and "end-date"
+  function calculateTripDuration(start, end) {
+    if (start && end) {
+      const startObject = new Date(start); // transforms the "start" argument into Date format which is needed to make calculations with dates
+      const endObject = new Date(end);
+      const durationInMilliseconds = endObject - startObject; // Date calculation in javascript is by default done in milliseconds. In the next line I converted the milliseconds in days.
+      const durationInDays = Math.floor(
+        durationInMilliseconds / (1000 * 60 * 60 * 24) + 1 // the + 1 is needed to also count the day of the start date
+      );
+      setTripDurationInDays(durationInDays);
+    } else {
+      setTripDurationInDays(0);
+    }
+  }
```

Comment:
Very nice exactly that's the approach one should take best practice-wise when working with dates, always convert to the smallest possible in this case the milliseconds, afaik the approach I know about is `new Date().valueOf()` this gives you the timestamp value and then you can get a pure milisecond different from there too but as I understand by your comments the simple `new Date()` is already handling it that way so, superb! :D